stdlib/public/core/AnyHashable.swift

```diff
@@ -196,3 +209,24 @@ func _stdlib_makeAnyHashableUpcastingToHashableBaseType<H : Hashable>(
   storingResultInto result: UnsafeMutablePointer<AnyHashable>
 )
 
+@_silgen_name("_swift_convertToAnyHashable")
+public // COMPILER_INTRINSIC
+func _convertToAnyHashable<H : Hashable>(_ value: H) -> AnyHashable {
+  return AnyHashable(value)
+}
+
+@_silgen_name("_swift_convertToAnyHashableIndirect")
+public // COMPILER_INTRINSIC (actually, called from the runtime)
+func _convertToAnyHashableIndirect<H : Hashable>(
+  _ value: H,
+  _ target: UnsafeMutablePointer<AnyHashable>) {
+  target.initialize(to: AnyHashable(value))
+}
+
+@_silgen_name("_swift_anyHashableDownCastConditionalIndirect")
+public // COMPILER_INTRINSIC (actually, called from the runtime)
+func _anyHashableDownCastConditionalIndirect<T>(
+  _ value: UnsafePointer<AnyHashable>,
+  _ target: UnsafeMutablePointer<T>) -> Bool {
+  return value.pointee._downCastConditional(into: target)
```

Comment:
I see, this way you are potentially saving a heap box allocation for the container returned from `value.pointee`.  If that is the important part here, I would appreciate a comment about this!